scipy/sparse/linalg/dsolve/linsolve.py

```diff
@@ -177,7 +177,13 @@ def spsolve(A, b, permc_spec=None, use_umfpack=True):
             raise ValueError("convert matrix data to double, please, using"
                   " .astype(), or set linsolve.useUmfpack = False")
 
-        umf = umfpack.UmfpackContext(_get_umf_family(A))
+        umf_family = _get_umf_family(A)
```

Comment:
Because this repeats, maybe better to move it to `_get_umf_family`, e.g. `umf_family, A = _get_umf_family(A)`.

Reply:
OK. I also moved the test setup to a function.